File: assignments/assignment-1-solution/src/CrazyEights/Domain/PlayableCard.cs
using CrazyEights.Cards;

namespace CrazyEights.Domain;

/// <summary>
/// Wraps a card with the reason it is playable.
/// </summary>
public class PlayableCard
{
    public PlayableCard(ICard card, string playableReason)
    {
        Card = card;
        PlayableReason = playableReason;
    }

    public ICard Card { get; }

    public string PlayableReason { get; }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/domain/PlayableCard.java
package edu.kennesaw.crazy8s.domain;

import edu.kennesaw.crazy8s.cards.Card;

/**
 * Wraps a card with the reason it is playable.
 */
public class PlayableCard {
    private final Card card;
    private final String playableReason;

    public PlayableCard(Card card, String playableReason) {
        this.card = card;
        this.playableReason = playableReason;
    }

    public Card getCard() {
        return card;
    }

    public String getPlayableReason() {
        return playableReason;
    }
}
